stages/org.osbuild.zip

```diff
@@ -0,0 +1,91 @@
+#!/usr/bin/python3
+"""
+Compresses an input file using zip.
+"""
+
+import os
+import subprocess
+import sys
+
+import osbuild.api
+
+CAPABILITIES = ["CAP_MAC_ADMIN"]
+
+
+SCHEMA_2 = """
+"options": {
+  "additionalProperties": false,
+  "required": ["filename"],
+  "properties": {
+    "filename": {
+      "description": "Filename for tar archive",
+      "type": "string"
+    },
+    "paths": {
+      "type": "array",
+      "items": {
+        "type": "string",
+        "description": "If specified, the archive will only contain the specified paths, in the specified order"
+      }
+    },
+    "level": {
+      "description": "Compression level",
+      "type": "integer",
+      "minimum": 0,
+      "maximum": 9,
+      "default": 6
+    }
+  }
+},
+"inputs": {
+  "type": "object",
+  "additionalProperties": false,
+  "required": ["tree"],
+  "properties": {
+    "tree": {
+      "type": "object",
+      "additionalProperties": true
+    }
+  }
+}
+"""
+
+
+def main(inputs, output_dir, options):
+    tree = inputs["tree"]["path"]
```

Comment:
Do we always want it to be a tree? Could it be a mount too?

Reply:
Hm... not too sure if I can comment on if we need to accept a mount as input. I've been modelling this `zip` stage against the existing `org.osbuild.tar` stage, which only takes in tree as input: https://github.com/osbuild/osbuild/blob/main/stages/org.osbuild.tar#L93-L100

Reply:
i would just keep it the same as tar for now and we'll open new PRs if we ever need that.